tmail-backend/tmail-third-party/ai-bot/src/main/java/com/linagora/tmail/mailet/rag/RagListener.java

```diff
@@ -0,0 +1,116 @@
+/********************************************************************
+ *  As a subpart of Twake Mail, this file is edited by Linagora.    *
+ *                                                                  *
+ *  https://twake-mail.com/                                         *
+ *  https://linagora.com                                            *
+ *                                                                  *
+ *  This file is subject to The Affero Gnu Public License           *
+ *  version 3.                                                      *
+ *                                                                  *
+ *  https://www.gnu.org/licenses/agpl-3.0.en.html                   *
+ *                                                                  *
+ *  This program is distributed in the hope that it will be         *
+ *  useful, but WITHOUT ANY WARRANTY; without even the implied      *
+ *  warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR         *
+ *  PURPOSE. See the GNU Affero General Public License for          *
+ *  more details.                                                   *
+ ********************************************************************/
+package com.linagora.tmail.mailet.rag;
+
+import java.util.Set;
+
+import org.apache.james.core.Username;
+import org.apache.james.events.Event;
+import org.apache.james.events.EventListener;
+import org.apache.james.mailbox.MailboxManager;
+import org.apache.james.mailbox.MailboxSession;
+import org.apache.james.mailbox.MessageIdManager;
+import org.apache.james.mailbox.MessageManager;
+import org.apache.james.mailbox.Role;
+import org.apache.james.mailbox.SystemMailboxesProvider;
+import org.apache.james.mailbox.events.MailboxEvents;
+import org.apache.james.mailbox.model.FetchGroup;
+import org.apache.james.mailbox.model.MessageResult;
+import org.apache.james.mime4j.dom.Message;
+import org.apache.james.mime4j.message.DefaultMessageBuilder;
+import org.apache.james.mime4j.stream.MimeConfig;
+import org.apache.james.util.ReactorUtils;
+import org.apache.james.util.mime.MessageContentExtractor;
+import org.reactivestreams.Publisher;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.inject.Inject;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+
+public class RagListener implements EventListener.ReactiveEventListener {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RagListener.class);
+
+    private final MailboxManager mailboxManager;
+    private final MessageIdManager messageIdManager;
+    private final SystemMailboxesProvider systemMailboxesProvider;
+    private final Set<Username> whitelist;
```

Comment:
```suggestion
    private final Optional<Set<Username>> whitelist;
```

To handle the case "no whitelist == this is reported to the RAG".

BTW this edge case is not tested...